vm/hash.go

```diff
@@ -200,8 +689,12 @@ func (h *HashObject) toString() string {
 	var out bytes.Buffer
 	var pairs []string
 
-	for key, value := range h.Pairs {
-		pairs = append(pairs, fmt.Sprintf("%s: %s", key, value.toString()))
+	for _, key := range h.sortedKeys() {
+		if _, isString := h.Pairs[key].(*StringObject); isString {
```

Comment:
I don't like that we need to check this for every pair. But I don't have better idea to improve this yet.